### ISSUE TITLE
Adding topologySpreadConstraint to velero default helm values

### ISSUE THREAD

**Issue by srujansagi** [vmware-tanzu/velero#7478]
**Describe the problem/challenge you have**

Currently velero helm chart doesnt support `topologySpreadConstraint` by default, can we add this optional attribute to default values.

**Describe the solution you'd like**

As velero is a key component to take backups and restore Kubernetes clusters and its objects, having topologySpreadConstraint attribute will increase the resiliency and avoid single point of failure by spreading the velero pods across different nodes. So having `topologySpreadConstraint` as an option will give control make sure it is highly available and resiliant.

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by blackpiglet** [comment]
In the current design, the Velero server deployment only supports 1 replica, and the Velero node-agent is DaemonSet.
IMO, it seems no need to consider the topology spread.

**Comment by reasonerjt** [comment]
I think the comment from @blackpiglet makes sense 
@srujansagi any thoughts on that?

**Comment by srujansagi** [comment]
> In the current design, the Velero server deployment only supports 1 replica, and the Velero node-agent is DaemonSet. IMO, it seems no need to consider the topology spread.

Not sure what do you mean by deployment only supports 1 replica, i changed the deployment to have 3 replicas and it spin up 3 pods just fine.

![image](https://github.com/vmware-tanzu/velero/assets/32345211/986b3d46-6215-4924-a182-f69d4d304bfd)

I dont see any DaemoSet in velero namespace, did i mis-understood this? We install velero using helm chart from this repo https://vmware-tanzu.github.io/helm-charts and currently running chart version 5.0.0

![image](https://github.com/vmware-tanzu/velero/assets/32345211/9e80601c-7316-4069-a8b7-aa79a0a41649)

**Comment by blackpiglet** [comment]
@srujansagi 
Of course, you can scale the Velero deployment to more than 1 replica, but the Velero itself is customized to work in the multiple instances environment. In this case, multiple Velero instances can update the Velero Backup/Restore CR at the same time. That will cause chaos in the Velero backup/restore workflow.

Deploying one instance of Velero is always recommended by the current design.

The DaemonSet is the node-agent. Please check this link for more information. https://velero.io/docs/v1.13/customize-installation/

**Comment by srujansagi** [comment]
@blackpiglet That makes sense thank you for the explanation. Are there any upcoming plans to enable support for topologySpreadConstraints with multiple instances (pods) operating within the same cluster in Kubernetes?

Given the current recommended design of running a single instance of Velero, it's vital to address high availability (HA) concerns to prevent any single point of failure. For instance, if a Velero pod or its hosting node were to fail, it could potentially disrupt scheduled backups across multiple clusters.

**Comment by blackpiglet** [comment]
That's a reasonable concern, but, with the current design, even if the Velero starts to support multiple instances, the backup and restore cannot recover when the running node crashes.

There are discussions about making the Velero controllers support multiple reconciliations simultaneously, but making the backup and restore recoverable is still not in the roadmap, because it's hard to cover all the error cases that could happen, so the suggested solution is deprecating the failed backup or restore, and create a new one.
